templates/.kokoro/build.sh

```diff
@@ -23,38 +23,37 @@ cd ${scriptDir}/..
 # include common functions
 source ${scriptDir}/common.sh
 
-# Find the files changed from when the PR branched to the last commit
-# Filter for java modules and get all the unique elements
-# grep returns 1 (error code) and exits the pipeline if there is no match
-# If there is no match, it will return true so the rest of the commands can run
-modified_files=$(git diff --name-only $KOKORO_GITHUB_PULL_REQUEST_COMMIT $KOKORO_GITHUB_PULL_REQUEST_TARGET_BRANCH)
-printf "Modified files:\n%s\n" "${modified_files}"
-directories=$(echo "${modified_files}" | grep -e 'java-.*' || true)
-printf "Files in java modules:\n%s\n" "${directories}"
-if [[ -n $directories ]]; then
-  directories=$(echo "${directories}" | cut -d '/' -f1 | sort -u)
-  dir_list=()
-  for directory in $directories
-  do
-    dir_list+=($directory)
-  done
-  # Combine each entry with a comma
-  module_list=$(IFS=, ; echo "${dir_list[*]}")
-fi
-printf "Module List:\n%s\n" "${module_list}"
-
 # Use GCP Maven Mirror
 mkdir -p ${HOME}/.m2
 cp settings.xml ${HOME}/.m2
 
+function assign_modules_to_job() {
+  modules=$(mvn help:evaluate -Dexpression=project.modules | grep '<.*>.*</.*>' | sed -e 's/<.*>\(.*\)<\/.*>/\1/g')
+  module_list=()
+  num=0
+  for module in $modules
+  do
+    # Add 1 as JOB_NUMBER is 1-indexed instead of 0-indexed
+    mod_num=$((num % NUM_JOBS + 1))
```

Comment:
NUM_JOB is ENV_VAR passed in. We can create new .cfg files and update this value in the common build.proto